### DISCUSSION TITLE
How to deploy with README.md instead of index.html

### DISCUSSION THREAD

**Discussion by ThomasCarstens** [vercel/community#1179] [Help] [answered]
I have a github repo with a README.md but no index.html
I was wondering how to make my vercel deployment automatically display the README if no index file is present.
Couldn't get it to work in my tests or find adequate documentation.
Thanks!

Tested this by running a website with a simple index.html file (this works) but renaming it README.md

**Comment by amyegan** [comment, marked as answer]
Hi @ThomasCarstens! I think you would need to use some sort of framework to serve a README.md as the home page of your site. Maybe something like [Harp](http://harpjs.com/docs/development/markdown) or [Markserv](https://github.com/markserv/markserv).

Or, if you just want to display the raw markdown with no styling, you could rewrite with [Edge Middleware](https://vercel.com/docs/concepts/functions/edge-middleware#) to use the markdown contents in place place of index.html as the default page.

**Reply by ThomasCarstens** [reply]
Thank you, have never heard of these. I'm actually hoping to deploy without a framework. Seen as the difference is just the name of the default index page, maybe there is a way to change this default value? (environment variables on build step?)

**Reply by amyegan** [reply]
I was able to display the raw markdown contents as the default page for my test site using a vercel.json rewrite config. My test used only these two files:

vercel.json
```json
{
  "rewrites": [
    { "source": "/", "destination": "/README.md" }
  ]
}
```

readme.md
```
# Hello

This is a test page.
```
